src/Streamly/Internal/Data/Stream/IsStream/Transform.hs

```diff
@@ -896,6 +900,32 @@ nubBy :: -- (IsStream t, Monad m) =>
     (a -> a -> Bool) -> t m a -> t m a
 nubBy = undefined -- fromStreamD . D.nubBy . toStreamD
 
+{-# INLINE_NORMAL nubByMerge #-}
+nubByMerge :: (Monad m) => (a -> a -> Bool) -> D.Stream m a -> D.Stream m a
```

Comment:
1. This is the same as uniqBy if uniqBy is applied on a sorted stream. We do not want to keep two combinators doing the same thing.
2. If you think this implementation is faster, then replace `uniqBy` by this implementation, add a benchmark for uniqBy, and compare the results of the previous implementation with the new one. If this is faster we can keep this otherwise we will keep the old one as it is more idiomatic.

Reply:
I was talking about `uniqBy`, not `nubBy`.